tl.go

```diff
@@ -0,0 +1,177 @@
+}
+
+// verifyLogEntry verifies a transparency log entry
+func (tl *RekorTL) VerifyLogEntry(ctx context.Context, entryBytes []byte) error {
+	entry := new(models.LogEntryAnon)
+	err := entry.UnmarshalBinary(entryBytes)
+	if err != nil {
+		return fmt.Errorf("error failed to unmarshal TL entry: %w", err)
+	}
+	err = entry.Verification.Validate(strfmt.Default)
+	if err != nil {
+		return fmt.Errorf("TL entry failed validation: %w", err)
+	}
+
+	verifier, err := loadVerifier()
+	if err != nil {
+		return fmt.Errorf("error failed to load TL verifier: %w", err)
+	}
+	err = verify.VerifyLogEntry(ctx, entry, verifier)
+	if err != nil {
+		return fmt.Errorf("TL entry failed verification: %w", err)
+	}
+	return nil
+}
+
+func loadVerifier() (signature.Verifier, error) {
+	rekorClient, err := rclient.GetRekorClient(DefaultRekorURL)
+	if err != nil {
+		return nil, fmt.Errorf("error creating rekor client: %w", err)
+	}
+	// fetch key from server
+	keyResp, err := rekorClient.Pubkey.GetPublicKey(nil)
+	if err != nil {
+		return nil, err
+	}
+	publicKey := keyResp.Payload
+
+	block, _ := pem.Decode([]byte(publicKey))
+	if block == nil {
+		return nil, errors.New("failed to decode public key of server")
+	}
+
+	pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return signature.LoadVerifier(pub, crypto.SHA256)
+}
+
+// CreateX509Cert generates a self-signed x509 cert from a PK token
+func CreateX509Cert(pkToken *pktoken.PKToken, signer crypto.Signer) ([]byte, error) {
+	pkTokenJSON, err := json.Marshal(pkToken)
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling PK token to JSON: %w", err)
+	}
+
+	// get subject identitifer from pk token
+	var payload struct {
+		Subject string `json:"sub"`
+	}
+	if err := json.Unmarshal(pkToken.Payload, &payload); err != nil {
+		return nil, err
+	}
+
+	// encode ephemeral public key
+	ecPub, err := x509.MarshalPKIXPublicKey(signer.Public())
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling public key: %w", err)
+	}
+
+	template := x509.Certificate{
+		SerialNumber:            big.NewInt(1),
+		Subject:                 pkix.Name{CommonName: payload.Subject},
+		RawSubjectPublicKeyInfo: ecPub,
+		NotBefore:               time.Now(),
+		NotAfter:                time.Now().Add(365 * 24 * time.Hour), // valid for 1 year
+		KeyUsage:                x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:             []x509.ExtKeyUsage{x509.ExtKeyUsageCodeSigning},
+		BasicConstraintsValid:   true,
+		DNSNames:                []string{payload.Subject},
+		IsCA:                    false,
+		SubjectKeyId:            pkTokenJSON,
+	}
+
+	// create a self-signed X.509 certificate
+	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, signer.Public(), signer)
+	if err != nil {
+		return nil, fmt.Errorf("error creating X.509 certificate: %w", err)
+	}
+	certBlock := &pem.Block{Type: "CERTIFICATE", Bytes: certDER}
+	return pem.EncodeToMemory(certBlock), nil
+}
```

Comment:
Is it possible to add tests for these functions?

Reply:
yes, added tests for cert creation [2a98027](https://github.com/openpubkey/signed-attestation/pull/4/commits/2a9802787a6d5d64243b01cda6278800221853ca)
the others are mostly rekor/cosign so not easily tested here